database/factories/UserFactory.php

```diff
@@ -1,26 +1,33 @@
 <?php
 
-/** @var \Illuminate\Database\Eloquent\Factory $factory */
+declare(strict_types=1);
+
+namespace Database\Factories;
 
 use App\Models\User;
-use Faker\Generator as Faker;
+use Illuminate\Database\Eloquent\Factories\Factory;
 use Illuminate\Support\Str;
 
-/*
-|--------------------------------------------------------------------------
-| Model Factories
-|--------------------------------------------------------------------------
-|
-| This directory should contain each of the model factory definitions for
-| your application. Factories provide a convenient way to generate new
-| model instances for testing / seeding your application's database.
-|
-*/
-$factory->define(User::class, function (Faker $faker) {
-    return [
-        'name'           => $faker->name,
-        'email'          => $faker->unique()->safeEmail,
-        'password'       => '$2y$10$OwXT/21pAv5OSIXyG1caxexYkL0uDJQSadZ4f46Y5AjmfSJtdkphu',
-        'remember_token' => Str::random(10),
-    ];
-});
+final class UserFactory extends Factory
+{
+    protected $model = User::class;
+
+    public function definition()
+    {
+        return [
+            'name'           => $this->faker->name,
+            'email'          => $this->faker->unique()->safeEmail,
+            'password'       => '$2y$10$OwXT/21pAv5OSIXyG1caxexYkL0uDJQSadZ4f46Y5AjmfSJtdkphu',
+            'remember_token' => Str::random(10),
+        ];
+    }
+
+    /**
+     * @param  array $attributes
+     * @return \Illuminate\Database\Eloquent\Model|\App\Models\User
+     */
+    public function createOne($attributes = [])
+    {
+        return parent::createOne($attributes);
```

Comment:
Dit lijkt overbodig, of is $attributes default niet een lege array?

Reply:
en btw: ik vind "createOne" nu niet echt een hele noodzakelijke verbetering t.o.v. create. Ja, het is zo duidelijk dat er 1 wordt gecreëerd en niet meer, maar dan nog....

Reply:
Tja dat is laravels keus haha, waarom ik deze method overschrijf (en ook in de stub) is zodat de return type ook `\App\Models\User` is, dan herkent hij de type in alle tests meteen, vond ik wel chill